src/main/java/lz/modelTesting/serivces/MongoUserDetailsService.java

```diff
@@ -0,0 +1,33 @@
+package lz.modelTesting.serivces;
+
+import lombok.Getter;
+import lombok.Setter;
+import lz.modelTesting.documents.User;
+import lz.modelTesting.repositories.UsersRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.List;
+
+@Component
+public class MongoUserDetailsService implements UserDetailsService {
+    @Getter
+    @Setter
+    @Autowired
+    private UsersRepository repository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        User user = repository.findByEmail(email);
+        if (user == null) {
+            throw new UsernameNotFoundException("User not found");
```

Comment:
when `findByEmail` returns Optional, it can be replaced with:
```
Optional<User> userOptional = repository.findByEmail(email);
User user = userOptional.orElseThrow(() -> new UsernameNotFoundException("User not found"));
```